From 1014c270361481a53ca3185c7c2f0ebc1c74d692 Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Mon, 5 Mar 2012 19:49:36 +0530
Subject: [PATCH 0840/1092] ARM: OMAP2+: gpmc: Handle clock by pm_runtime API

This patch updates clock API to pm_runtime API for GPMC clock
activity. This will use HWMOD data which is preferred upon regular
clk_[disable/enable].

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/mach-omap2/gpmc.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-omap2/gpmc.c b/arch/arm/mach-omap2/gpmc.c
index 8b968b2..d163b44 100644
--- a/arch/arm/mach-omap2/gpmc.c
+++ b/arch/arm/mach-omap2/gpmc.c
@@ -26,6 +26,7 @@
 #include <linux/io.h>
 #include <linux/module.h>
 #include <linux/interrupt.h>
+#include <linux/pm_runtime.h>
 
 #include <asm/mach-types.h>
 #include <plat/gpmc.h>
@@ -166,7 +167,7 @@ u32 gpmc_cs_read_reg(int cs, int idx)
 
 static struct clk *gpmc_l3_clk;
 
-static void __devinit gpmc_clk_init(void)
+static void __devinit gpmc_clk_init(struct device *dev)
 {
 	char *ck = NULL;
 
@@ -186,7 +187,8 @@ static void __devinit gpmc_clk_init(void)
 		BUG();
 	}
 
-	clk_enable(gpmc_l3_clk);
+	pm_runtime_enable(dev);
+	pm_runtime_get_sync(dev);
 }
 
 /* TODO: Add support for gpmc_fck to clock framework and use it */
@@ -753,7 +755,7 @@ static int __devinit gpmc_probe(struct platform_device *pdev)
 	void *p;
 
 	/* XXX: This should go away with HWMOD & runtime PM adaptation */
-	gpmc_clk_init();
+	gpmc_clk_init(&pdev->dev);
 
 	gpmc = devm_kzalloc(&pdev->dev, sizeof(struct gpmc), GFP_KERNEL);
 	if (!gpmc)
-- 
1.7.11.2

